qa/suites/orch/cephadm/upgrade/3-upgrade/staggered.yaml

```diff
@@ -0,0 +1,78 @@
+tasks:
+- cephadm.shell:
+    env: [sha1]
+    mon.a:
+      - radosgw-admin realm create --rgw-realm=r --default
+      - radosgw-admin zonegroup create --rgw-zonegroup=default --master --default
+      - radosgw-admin zone create --rgw-zonegroup=default --rgw-zone=z --master --default
+      - radosgw-admin period update --rgw-realm=r --commit
+      - ceph orch apply rgw r z --placement=2 --port=8000
+      - sleep 120
+      - ceph config set mon mon_warn_on_insecure_global_id_reclaim false --force
+      - ceph config set mon mon_warn_on_insecure_global_id_reclaim_allowed false --force
+      - ceph config set global log_to_journald false --force
+      # get some good info on the state of things pre-upgrade. Useful for debugging
+      - ceph orch ps
+      - ceph versions
+      - ceph -s
+      - ceph orch ls
+      # doing staggered upgrade requires mgr daemons being on a version that contains the staggered upgrade code
+      # until there is a stable version that contains it, we can test by manually upgrading a mgr daemon
+      - ceph config set mgr container_image quay.ceph.io/ceph-ci/ceph:$sha1
+      - ceph orch daemon redeploy "mgr.$(ceph mgr dump -f json | jq .standbys | jq .[] | jq -r .name)"
+      - ceph orch ps --refresh
+      - sleep 60
+      - ceph orch ps
+      - ceph versions
+      - ceph -s
+      - ceph versions | jq -e '.mgr | length == 2'
+      - ceph mgr fail
+      - sleep 60
+      - ceph orch daemon redeploy "mgr.$(ceph mgr dump -f json | jq .standbys | jq .[] | jq -r .name)"
+      - ceph orch ps --refresh
+      - sleep 60
+      - ceph orch ps
+      - ceph versions
+      - ceph -s
+      - ceph versions | jq -e '.mgr | length == 1'
+      - ceph mgr fail
+      - sleep 60
+      - ceph orch ps
+      - ceph versions
+      - ceph orch upgrade start --image quay.ceph.io/ceph-ci/ceph:$sha1 --daemon-types mgr
+      - while ceph orch upgrade status | jq '.in_progress' | grep true && ! ceph orch upgrade status | jq '.message' | grep Error ; do ceph orch ps ; ceph versions ; ceph orch upgrade status ; sleep 30 ; done
+      - ceph versions | jq -e '.mgr | length == 1'
+      - ceph versions | jq -e '.mgr | keys' | grep $sha1
+      - ceph versions | jq -e '.overall | length == 2'
+      - ceph orch upgrade check quay.ceph.io/ceph-ci/ceph:$sha1 | jq -e '.up_to_date | length == 2'
```

Comment:
For all the similar checks it would be good to indicate what are we checking for (why are we expecting a `length == 2`).

Reply:
added a bunch of comments to this test to explain what each part is checking